proposals/gc/MVP.md

```diff
@@ -325,39 +325,44 @@ Tentatively, support a type of guaranteed unboxed scalars.
   - `ref.is_i31 : [anyref] -> [i32]`
 
 * `br_on_func <labelidx>` branches if a reference is a function
-  - `br_on_func $l : [anyref] -> [anyref]`
-    - iff `$l : [funcref]`
+  - `br_on_func $l : [(ref null ht)] -> [(ref null ht)]`
```

Comment:
I think the typing of all `br_on_*` instructions, including `br_on_cast`, could be further refined to preserve their argument's nullability, similarly to `ref.cast`.

Reply:
Good point, PTAL!